diff --git a/crates/syntax/fuzz/Cargo.toml b/crates/syntax/fuzz/Cargo.toml
index c2c6dac..206daac 100644
--- a/crates/syntax/fuzz/Cargo.toml
+++ b/crates/syntax/fuzz/Cargo.toml
@@ -10,7 +10,6 @@ cargo-fuzz = true
 
 [dependencies]
 syntax = { path = "..", version = "0.0.0" }
-text-edit = { path = "../../text-edit", version = "0.0.0" }
 libfuzzer-sys = "0.4.5"
 
 # Prevent this from interfering with workspaces
@@ -25,5 +24,3 @@ path = "fuzz_targets/parser.rs"
 name = "reparse"
 path = "fuzz_targets/reparse.rs"
 
-[lints]
-workspace = true
