diff --git a/fuzz/fuzz_targets/compress.rs b/fuzz/fuzz_targets/compress.rs
index 79d36bf..2362b0d 100644
--- a/fuzz/fuzz_targets/compress.rs
+++ b/fuzz/fuzz_targets/compress.rs
@@ -5,9 +5,8 @@ use zlib_rs::ReturnCode;
 
 fuzz_target!(|data: String| {
     // first, deflate the data using the standard zlib
-    let length = 8 * 1024;
+    let mut length = 8 * 1024;
     let mut deflated = vec![0; length as usize];
-    let mut length = length as u64;
     let error = unsafe {
         libz_rs_sys::compress(
             deflated.as_mut_ptr().cast(),
