diff --git a/fuzz-wasm/Cargo.toml b/fuzz-wasm/Cargo.toml
new file mode 100644
index 00000000..3371f2ff
--- /dev/null
+++ b/fuzz-wasm/Cargo.toml
@@ -0,0 +1,19 @@
+[package]
+name = "zip2-fuzz-wasm"
+version = "0.0.0"
+edition = "2021"
+publish = false
+
+[package.metadata]
+cargo-fuzz = true
+
+[workspace]
+members = ["."]
+
+[[bin]]
+name = "zip2-read"
+path = "src/main.rs"
+
+[dependencies]
+libfuzzer-sys = "0.4"
+zip = { path = "..", default-features = false, features = ["deflate"] }
diff --git a/fuzz-wasm/src/main.rs b/fuzz-wasm/src/main.rs
new file mode 100644
index 00000000..4d458769
--- /dev/null
+++ b/fuzz-wasm/src/main.rs
@@ -0,0 +1,16 @@
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+use std::io::Read;
+
+fuzz_target!(|data: &[u8]| {
+    let reader = std::io::Cursor::new(data);
+    if let Ok(mut archive) = zip::ZipArchive::new(reader) {
+        for i in 0..archive.len() {
+            if let Ok(mut file) = archive.by_index(i) {
+                let mut sink = Vec::new();
+                let _ = file.read_to_end(&mut sink);
+            }
+        }
+    }
+});
