diff --git a/pngerror.c b/pngerror.c
index ec3a709b9..a33f473fb 100644
--- a/pngerror.c
+++ b/pngerror.c
@@ -751,7 +751,14 @@ png_default_error,(png_const_structrp png_ptr, png_const_charp error_message),
 #else
    PNG_UNUSED(error_message) /* Make compiler happy */
 #endif
+#ifndef __wasi__
    png_longjmp(png_ptr, 1);
+#else
+   void wasmfuzz_exit_testcase()
+      __attribute__((__noreturn__))
+      __attribute__((__import_module__("wasmfuzz"), __import_name__("exit_testcase")));
+   wasmfuzz_exit_testcase();
+#endif
 }
 
 PNG_FUNCTION(void,PNGAPI
