diff --git a/programs/fuzz/Makefile b/programs/fuzz/Makefile
index 65ac6f8949..1a8419fec4 100644
--- a/programs/fuzz/Makefile
+++ b/programs/fuzz/Makefile
@@ -1,5 +1,7 @@
 MBEDTLS_TEST_PATH:=../../tests
 
+CFLAGS ?= -g -O2 -flto
+
 MBEDTLS_PATH := ../..
 include ../../scripts/common.make
 
@@ -7,9 +9,7 @@ PROGRAM_FUZZ_PATH:=$(MBEDTLS_PATH)/tf-psa-crypto/programs/fuzz
 
 DEP=${MBEDLIBS}
 
-ifdef FUZZINGENGINE
-LOCAL_LDFLAGS += -lFuzzingEngine
-endif
+LOCAL_LDFLAGS += ${FUZZ_LDFLAGS}
 
 LOCAL_CFLAGS += -I$(PROGRAM_FUZZ_PATH)
 
