diff --git a/platform/platform_util.c b/platform/platform_util.c
index 3dd58f3aa..f2eecd2a2 100644
--- a/platform/platform_util.c
+++ b/platform/platform_util.c
@@ -237,7 +237,10 @@ mbedtls_ms_time_t mbedtls_ms_time(void)
     return current_ms;
 }
 #else
-#error "No mbedtls_ms_time available"
+mbedtls_ms_time_t mbedtls_ms_time(void)
+{
+    return 0;
+}
 #endif
 #endif /* MBEDTLS_HAVE_TIME && !MBEDTLS_PLATFORM_MS_TIME_ALT */
 
@@ -245,7 +248,7 @@ mbedtls_ms_time_t mbedtls_ms_time(void)
 
 #if !defined(MBEDTLS_PLATFORM_IS_UNIXLIKE) && \
     !defined(__MVS__) /* z/OS */ &&           \
-    !defined(_WIN32)
+    !defined(_WIN32) && !defined(__wasm__)
 #error \
     "The built-in entropy sources only work on Unix and Windows. " \
     "Please enable MBEDTLS_PSA_DRIVER_GET_ENTROPY instead of " \
