diff --git a/drivers/builtin/src/entropy_poll.c b/drivers/builtin/src/entropy_poll.c
index bd2cf693e..9b57d85dd 100644
--- a/drivers/builtin/src/entropy_poll.c
+++ b/drivers/builtin/src/entropy_poll.c
@@ -29,9 +29,11 @@
 
 #if !defined(MBEDTLS_NO_PLATFORM_ENTROPY)
 
+
 #if !defined(unix) && !defined(__unix__) && !defined(__unix) && \
     !defined(__APPLE__) && !defined(_WIN32) && !defined(__QNXNTO__) && \
-    !defined(__HAIKU__) && !defined(__midipix__) && !defined(__MVS__)
+    !defined(__HAIKU__) && !defined(__midipix__) && !defined(__MVS__) && \
+    !defined(__wasm__)
 #error \
     "Platform entropy sources only work on Unix and Windows, see MBEDTLS_NO_PLATFORM_ENTROPY in mbedtls_config.h"
 #endif
@@ -110,6 +112,19 @@ static int getrandom_wrapper(void *buf, size_t buflen, unsigned int flags)
           (__DragonFly__ && __DragonFly_version >= 500700) */
 #endif /* __FreeBSD__ || __DragonFly__ */
 
+#if defined(__wasm__)
+#include <errno.h>
+#include <sys/random.h>
+#define HAVE_GETRANDOM
+static int getrandom_wrapper(void *buf, size_t buflen, unsigned int flags)
+{
+    (void) flags;
+    if (!getentropy(buf, buflen))
+        return buflen;
+    return 0;
+}
+#endif
+
 /*
  * Some BSD systems provide KERN_ARND.
  * This is equivalent to reading from /dev/urandom, only it doesn't require an
diff --git a/drivers/builtin/src/platform_util.c b/drivers/builtin/src/platform_util.c
index 19ef07aea..ecf6d08a3 100644
--- a/drivers/builtin/src/platform_util.c
+++ b/drivers/builtin/src/platform_util.c
@@ -260,4 +260,11 @@ mbedtls_ms_time_t mbedtls_ms_time(void)
 #else
 #error "No mbedtls_ms_time available"
 #endif
+
+#else
+mbedtls_ms_time_t mbedtls_ms_time(void)
+{
+    return 0;
+}
+
 #endif /* MBEDTLS_HAVE_TIME && !MBEDTLS_PLATFORM_MS_TIME_ALT */
