diff --git a/lib/codebook.c b/lib/codebook.c
--- a/lib/codebook.c
+++ b/lib/codebook.c
@@ -407,6 +407,9 @@ long vorbis_book_decode(dec_codebook *book, oggpack_buffer *b){
 long vorbis_book_decodevs_add(dec_codebook *book,float *a,oggpack_buffer *b,
                               int n){
   if(book->codelist){
+    if(book->dim <= 0){
+      return(-1);
+    }
     int step=n/book->dim;
     long *entry = alloca(sizeof(*entry)*step);
     float **t = alloca(sizeof(*t)*step);
