diff --git a/crates/native-c/src/demangle.c b/crates/native-c/src/demangle.c
index e4eb399..cde9cfe 100644
--- a/crates/native-c/src/demangle.c
+++ b/crates/native-c/src/demangle.c
@@ -111,6 +111,7 @@ struct printer {
 
 static NODISCARD overflow_status printer_print_path(struct printer *printer, bool in_value);
 static NODISCARD overflow_status printer_print_type(struct printer *printer);
+static NODISCARD overflow_status printer_print_pat(struct printer *printer);
 static NODISCARD overflow_status printer_print_const(struct printer *printer, bool in_value);
 
 static NODISCARD demangle_status try_parse_path(struct parser *parser) {
@@ -1662,6 +1663,11 @@ static NODISCARD overflow_status printer_print_type(struct printer *printer) {
     case 'B':
         PRINT(printer_print_backref(printer, printer_print_type_backref, NULL));
         break;
+    case 'W':
+        PRINT(printer_print_type(printer));
+        PRINT_STR(printer, " is ");
+        PRINT(printer_print_pat(printer));
+        break;
     default:
         // Go back to the tag, so `print_path` also sees it.
         if (printer->status == DemangleOk && /* safety */ printer->parser.next > 0) {
@@ -1674,6 +1680,39 @@ static NODISCARD overflow_status printer_print_type(struct printer *printer) {
     return OverflowOk;
 }
 
+static NODISCARD overflow_status printer_print_pat(struct printer *printer) {
+    uint8_t tag;
+    PARSE(printer, parser_ch, &tag);
+
+    switch(tag) {
+    case 'R':
+        PRINT(printer_print_const(printer, false));
+        PRINT_STR(printer, "..=");
+        PRINT(printer_print_const(printer, false));
+        break;
+    case 'O':
+        PARSE(printer, parser_push_depth);
+        PRINT(printer_print_pat(printer));
+        while (!printer_eat(printer, 'E')) {
+            // Avoid endless loops if we reached the end while parsing.
+            if (printer->status != DemangleOk) {
+                INVALID(printer);
+            }
+            PRINT_STR(printer, " | ");
+            PRINT(printer_print_pat(printer));
+        }
+        printer_pop_depth(printer);
+        break;
+    case 'N':
+        PRINT_STR(printer, "!null");
+        break;
+    default:
+        INVALID(printer);
+    }
+
+    return OverflowOk;
+}
+
 NODISCARD static demangle_status rust_demangle_legacy_demangle(const char *s, size_t s_len, struct demangle_legacy *res, const char **rest)
 {
     if (s_len > strlen(s)) {
@@ -1846,8 +1885,17 @@ NODISCARD static overflow_status rust_demangle_legacy_display_demangle(struct de
                     if (escape_len > 1 && escape_start[0] == 'u') {
                         escape_start++;
                         escape_len--;
+                        bool all_lower_hex = true;
+                        for (size_t i = 0; i < escape_len; i++) {
+                            char hex = escape_start[i];
+                            if (!((hex >= '0' && hex <= '9') || (hex >= 'a' && hex <= 'f'))) {
+                                all_lower_hex = false;
+                                break;
+                            }
+                        }
                         uint64_t val;
-                        if (try_parse_uint(escape_start, escape_len, &val)
+                        if (all_lower_hex
+                            && try_parse_uint(escape_start, escape_len, &val)
                             && val < UINT32_MAX
                             && validate_char((uint32_t)val))
                         {
