diff --git a/src/lib_json/json_value.cpp b/src/lib_json/json_value.cpp
index 168251e..0ce4ab7 100644
--- a/src/lib_json/json_value.cpp
+++ b/src/lib_json/json_value.cpp
@@ -214,13 +214,17 @@ JSONCPP_NORETURN void throwLogicError(String const& msg) {
   throw LogicError(msg);
 }
 #else // !JSON_USE_EXCEPTION
+extern "C" void wasmfuzz_exit_testcase()
+    __attribute__((__noreturn__))
+    __attribute__((__import_module__("wasmfuzz"), __import_name__("exit_testcase")));
+
 JSONCPP_NORETURN void throwRuntimeError(String const& msg) {
   std::cerr << msg << std::endl;
-  abort();
+  wasmfuzz_exit_testcase();
 }
 JSONCPP_NORETURN void throwLogicError(String const& msg) {
   std::cerr << msg << std::endl;
-  abort();
+  wasmfuzz_exit_testcase();
 }
 #endif
 
diff --git a/src/test_lib_json/fuzz.cpp b/src/test_lib_json/fuzz.cpp
index 3679a95..8367d3b 100644
--- a/src/test_lib_json/fuzz.cpp
+++ b/src/test_lib_json/fuzz.cpp
@@ -41,14 +41,7 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
 
   Json::Value root;
   const auto data_str = reinterpret_cast<const char*>(data);
-#if JSON_USE_EXCEPTION
-  try {
-#endif // JSON_USE_EXCEPTION
-    reader->parse(data_str, data_str + size, &root, nullptr);
-#if JSON_USE_EXCEPTION
-  } catch (Json::Exception const&) {
-  }
-#endif // JSON_USE_EXCEPTION
+  reader->parse(data_str, data_str + size, &root, nullptr);
   // Whether it succeeded or not doesn't matter.
   return 0;
 }
