diff --git a/libarchive/archive_read_disk_posix.c b/libarchive/archive_read_disk_posix.c
index a3c9e58..9af54d4 100644
--- a/libarchive/archive_read_disk_posix.c
+++ b/libarchive/archive_read_disk_posix.c
@@ -97,8 +97,10 @@
 #include "archive_private.h"
 #include "archive_read_disk_private.h"
 
+static int fchdir(int fd) { abort(); }
+
 #ifndef HAVE_FCHDIR
-#error fchdir function required.
+// #error fchdir function required.
 #endif
 #ifndef O_BINARY
 #define O_BINARY	0
diff --git a/libarchive/archive_read_disk_set_standard_lookup.c b/libarchive/archive_read_disk_set_standard_lookup.c
index ecd675d..0089879 100644
--- a/libarchive/archive_read_disk_set_standard_lookup.c
+++ b/libarchive/archive_read_disk_set_standard_lookup.c
@@ -232,15 +232,7 @@ lookup_uname_helper(struct name_cache *cache, id_t id)
 static const char *
 lookup_uname_helper(struct name_cache *cache, id_t id)
 {
-	struct passwd	*result;
-	(void)cache; /* UNUSED */
-
-	result = getpwuid((uid_t)id);
-
-	if (result == NULL)
-		return (NULL);
-
-	return strdup(result->pw_name);
+	return (NULL);
 }
 #endif
 
@@ -300,15 +292,7 @@ lookup_gname_helper(struct name_cache *cache, id_t id)
 static const char *
 lookup_gname_helper(struct name_cache *cache, id_t id)
 {
-	struct group	*result;
-	(void)cache; /* UNUSED */
-
-	result = getgrgid((gid_t)id);
-
-	if (result == NULL)
-		return (NULL);
-
-	return strdup(result->gr_name);
+	return (NULL);
 }
 #endif
 
diff --git a/libarchive/archive_read_support_filter_program.c b/libarchive/archive_read_support_filter_program.c
index 3a6eea5..32a98a6 100644
--- a/libarchive/archive_read_support_filter_program.c
+++ b/libarchive/archive_read_support_filter_program.c
@@ -238,6 +238,7 @@ program_bidder_bid(struct archive_read_filter_bidder *b,
 static int
 child_stop(struct archive_read_filter *f, struct program *program)
 {
+#if !defined(__wasi__)
 	/* Close our side of the I/O with the child. */
 	if (program->child_stdin != -1) {
 		close(program->child_stdin);
@@ -293,7 +294,7 @@ child_stop(struct archive_read_filter *f, struct program *program)
 		    WEXITSTATUS(program->exit_status));
 		return (ARCHIVE_WARN);
 	}
-
+#endif
 	return (ARCHIVE_WARN);
 }
 
@@ -399,6 +400,10 @@ program_reader_vtable = {
 int
 __archive_read_program(struct archive_read_filter *f, const char *cmd)
 {
+	#if defined(__wasi__)
+	// We can't run programs but don't want this to fail our whole thing.
+	return (ARCHIVE_WARN);
+	#endif
 	struct program *program;
 	static const size_t out_buf_len = 65536;
 	char *out_buf;
diff --git a/libarchive/archive_umask.c b/libarchive/archive_umask.c
index e66c9ac..29c5c72 100644
--- a/libarchive/archive_umask.c
+++ b/libarchive/archive_umask.c
@@ -46,6 +46,9 @@
 mode_t
 __archive_get_umask(void)
 {
+#ifdef __wasi__
+	return 0;
+#else
 	mode_t mask;
 
 #ifdef __linux__
@@ -91,4 +94,5 @@ __archive_get_umask(void)
 
 	umask(mask = umask(0));
 	return (mask);
+#endif
 }
diff --git a/libarchive/archive_write_add_filter_program.c b/libarchive/archive_write_add_filter_program.c
index 6b29018..29bf755 100644
--- a/libarchive/archive_write_add_filter_program.c
+++ b/libarchive/archive_write_add_filter_program.c
@@ -330,6 +330,7 @@ int
 __archive_write_program_close(struct archive_write_filter *f,
     struct archive_write_program_data *data)
 {
+#if !defined(__wasi__)
 	int ret, status;
 	pid_t pid;
 	ssize_t bytes_read;
@@ -386,6 +387,8 @@ cleanup:
 		ret = ARCHIVE_FATAL;
 	}
 	return ret;
+#endif
+	return ARCHIVE_WARN;
 }
 
 static void
diff --git a/libarchive/archive_write_set_format_iso9660.c b/libarchive/archive_write_set_format_iso9660.c
index fe5beed..5997c7b 100644
--- a/libarchive/archive_write_set_format_iso9660.c
+++ b/libarchive/archive_write_set_format_iso9660.c
@@ -2525,7 +2525,7 @@ get_tmfromtime(struct tm *tm, time_t *t)
 #if HAVE_LOCALTIME_S
 	localtime_s(tm, t);
 #elif HAVE_LOCALTIME_R
-# ifdef HAVE_TZSET
+# if defined(HAVE_TZSET) && !defined(__wasi__)
 	tzset();
 # endif
 	localtime_r(t, tm);
@@ -5189,8 +5189,8 @@ isoent_create_virtual_dir(struct archive_write *a, struct iso9660 *iso9660, cons
 	archive_entry_unset_mtime(file->entry);
 	archive_entry_unset_atime(file->entry);
 	archive_entry_unset_ctime(file->entry);
-	archive_entry_set_uid(file->entry, getuid());
-	archive_entry_set_gid(file->entry, getgid());
+	// archive_entry_set_uid(file->entry, getuid());
+	// archive_entry_set_gid(file->entry, getgid());
 	archive_entry_set_mode(file->entry, 0555 | AE_IFDIR);
 	archive_entry_set_nlink(file->entry, 2);
 	if (isofile_gen_utility_names(a, file) < ARCHIVE_WARN) {
@@ -7166,8 +7166,8 @@ isoent_create_boot_catalog(struct archive_write *a, struct isoent *rootent)
 	archive_entry_set_mtime(file->entry, iso9660->birth_time, 0);
 	archive_entry_set_atime(file->entry, iso9660->birth_time, 0);
 	archive_entry_set_ctime(file->entry, iso9660->birth_time, 0);
-	archive_entry_set_uid(file->entry, getuid());
-	archive_entry_set_gid(file->entry, getgid());
+	// archive_entry_set_uid(file->entry, getuid());
+	// archive_entry_set_gid(file->entry, getgid());
 	archive_entry_set_mode(file->entry, AE_IFREG | 0444);
 	archive_entry_set_nlink(file->entry, 1);

