--- a/src/decoder/transform/palette.rs
+++ b/src/decoder/transform/palette.rs
@@ -70,8 +70,8 @@ fn create_rgba_palette(info: &Info) -> [[u8; 4]; 256] {
             palette_iter = &palette_iter[3..];
             rgba_iter = &mut rgba_iter[1..];
         }
-        if !palette_iter.is_empty() {
+        if palette_iter.len() >= 3 {
             rgba_iter[0][0..3].copy_from_slice(&palette_iter[0..3]);
         }
     }
