diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33e4fa8..8789403 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -829,9 +829,6 @@ if(ENABLE_STATIC)
     src/transupp.c)
   target_link_libraries(jpegtran-static jpeg-static)
   set_property(TARGET jpegtran-static PROPERTY COMPILE_FLAGS "${USE_SETMODE}")
-
-  add_executable(example-static src/example.c)
-  target_link_libraries(example-static jpeg-static)
 endif()
 
 add_executable(rdjpgcom src/rdjpgcom.c)
