diff --git a/openpgp/fuzz/fuzz_targets/cert_from_bytes.rs b/openpgp/fuzz/fuzz_targets/cert_from_bytes.rs
index 905095d..f901040 100644
--- a/openpgp/fuzz/fuzz_targets/cert_from_bytes.rs
+++ b/openpgp/fuzz/fuzz_targets/cert_from_bytes.rs
@@ -8,9 +8,14 @@ use openpgp::{
     parse::Parse,
 };
 
+fuzz_target!(|data: &[u8]| {
+    let _ = Cert::from_bytes(data);
+});
+/*
 fuzz_target!(|data: &[u8]| -> Corpus {
     match Cert::from_bytes(data) {
         Ok(_) => Corpus::Keep,
         Err(_) => Corpus::Reject,
     }
 });
+*/
diff --git a/openpgp/src/parse.rs b/openpgp/src/parse.rs
index 6a8a1df..ed07026 100644
--- a/openpgp/src/parse.rs
+++ b/openpgp/src/parse.rs
@@ -1768,7 +1768,7 @@ impl Subpacket {
         php.field("subpacket length", length.serialized_len());
         let len = length.len() as usize;
 
-        if limit < length.serialized_len() + len {
+        if limit < length.serialized_len().saturating_add(len) {
             return Err(Error::MalformedPacket(
                 "Subpacket extends beyond the end of the subpacket area".into())
                        .into());
