diff --git a/fuzz/ml-dsa.c b/fuzz/ml-dsa.c
index 094af9096a..679a8116f6 100644
--- a/fuzz/ml-dsa.c
+++ b/fuzz/ml-dsa.c
@@ -10,6 +10,7 @@
 
 /* Test ML-DSA operation.  */
 #include <string.h>
+#include <stdio.h>
 #include <openssl/evp.h>
 #include <openssl/err.h>
 #include <openssl/rand.h>
diff --git a/fuzz/ml-kem.c b/fuzz/ml-kem.c
index 6e3aed8b23..c9275781f0 100644
--- a/fuzz/ml-kem.c
+++ b/fuzz/ml-kem.c
@@ -12,6 +12,7 @@
  * Test ml-kem operation.
  */
 #include <string.h>
+#include <stdio.h>
 #include <openssl/evp.h>
 #include <openssl/err.h>
 #include <openssl/rand.h>
diff --git a/fuzz/slh-dsa.c b/fuzz/slh-dsa.c
index 17238bcf8b..c969558611 100644
--- a/fuzz/slh-dsa.c
+++ b/fuzz/slh-dsa.c
@@ -12,6 +12,7 @@
  * Test slh-dsa operation.
  */
 #include <string.h>
+#include <stdio.h>
 #include <openssl/evp.h>
 #include <openssl/err.h>
 #include <openssl/rand.h>
