diff --git a/fuzz/ml-dsa.c b/fuzz/ml-dsa.c
index 1088f9d..ecec951 100644
--- a/fuzz/ml-dsa.c
+++ b/fuzz/ml-dsa.c
@@ -10,6 +10,7 @@
 
 /* Test ML-DSA operation.  */
 #include <string.h>
+#include <stdio.h>
 #include <openssl/evp.h>
 #include <openssl/err.h>
 #include <openssl/rand.h>
diff --git a/fuzz/ml-kem.c b/fuzz/ml-kem.c
index 7cf6e96..e8c4af4 100644
--- a/fuzz/ml-kem.c
+++ b/fuzz/ml-kem.c
@@ -12,6 +12,7 @@
  * Test ml-kem operation.
  */
 #include <string.h>
+#include <stdio.h>
 #include <openssl/evp.h>
 #include <openssl/err.h>
 #include <openssl/rand.h>
diff --git a/fuzz/slh-dsa.c b/fuzz/slh-dsa.c
index dada29a..24daa5e 100644
--- a/fuzz/slh-dsa.c
+++ b/fuzz/slh-dsa.c
@@ -12,6 +12,7 @@
  * Test slh-dsa operation.
  */
 #include <string.h>
+#include <stdio.h>
 #include <openssl/evp.h>
 #include <openssl/err.h>
 #include <openssl/rand.h>
