diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index c7002eff39..597bfac788 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -1953,6 +1953,17 @@ my %targets = (
         ranlib           => sub { env('RANLIB') },
     },
 
+##### WebAssembly via WASI
+    "wasm32-wasi" => {
+        inherit_from => ["BASE_unix"],
+        CC => "clang",
+        CXX => "clang++",
+        cflags => add("--target=wasm32-wasi"),
+        cxxflags => add("--target=wasm32-wasi"),
+        lib_cppflags => add("-DL_ENDIAN"),
+        bn_ops => "THIRTY_TWO_BIT",
+    },
+
     ##### VMS
     # Most things happen in vms-generic.
     # Note that vms_info extracts the pointer size from the end of
