commit c6e18603c6da62fb5ec19b4e820bb1e5fc4ebca1
Author: Mrmaxmeier <Mrmaxmeier@gmail.com>
Date:   Sat Jun 14 12:47:53 2025 +0200

    gix-index: fix wasi build

diff --git a/gix-index/src/fs.rs b/gix-index/src/fs.rs
index 2be45bb8d..32cd19da4 100644
--- a/gix-index/src/fs.rs
+++ b/gix-index/src/fs.rs
@@ -54,14 +54,14 @@ impl Metadata {
     pub fn modified(&self) -> Option<SystemTime> {
         #[cfg(not(windows))]
         {
-            #[cfg(not(any(target_os = "aix", target_os = "hurd")))]
+            #[cfg(not(any(target_os = "aix", target_os = "hurd", target_os = "wasi")))]
             let seconds = self.0.st_mtime;
-            #[cfg(any(target_os = "aix", target_os = "hurd"))]
+            #[cfg(any(target_os = "aix", target_os = "hurd", target_os = "wasi"))]
             let seconds = self.0.st_mtim.tv_sec;
 
-            #[cfg(not(any(target_os = "aix", target_os = "hurd")))]
+            #[cfg(not(any(target_os = "aix", target_os = "hurd", target_os = "wasi")))]
             let nanoseconds = self.0.st_mtime_nsec;
-            #[cfg(any(target_os = "aix", target_os = "hurd"))]
+            #[cfg(any(target_os = "aix", target_os = "hurd", target_os = "wasi"))]
             let nanoseconds = self.0.st_mtim.tv_nsec;
 
             // All operating systems treat the seconds as offset from unix epoch, hence it must
@@ -81,14 +81,14 @@ impl Metadata {
     pub fn created(&self) -> Option<SystemTime> {
         #[cfg(not(windows))]
         {
-            #[cfg(not(any(target_os = "aix", target_os = "hurd")))]
+            #[cfg(not(any(target_os = "aix", target_os = "hurd", target_os = "wasi")))]
             let seconds = self.0.st_ctime;
-            #[cfg(any(target_os = "aix", target_os = "hurd"))]
+            #[cfg(any(target_os = "aix", target_os = "hurd", target_os = "wasi"))]
             let seconds = self.0.st_ctim.tv_sec;
 
-            #[cfg(not(any(target_os = "aix", target_os = "hurd")))]
+            #[cfg(not(any(target_os = "aix", target_os = "hurd", target_os = "wasi")))]
             let nanoseconds = self.0.st_ctime_nsec;
-            #[cfg(any(target_os = "aix", target_os = "hurd"))]
+            #[cfg(any(target_os = "aix", target_os = "hurd", target_os = "wasi"))]
             let nanoseconds = self.0.st_ctim.tv_nsec;
 
             // All operating systems treat the seconds as offset from unix epoch, hence it must
