diff --git a/gix-path/src/convert.rs b/gix-path/src/convert.rs
index 23134b454..c93599929 100644
--- a/gix-path/src/convert.rs
+++ b/gix-path/src/convert.rs
@@ -7,10 +7,15 @@ use std::{
 use gix_error::ExnMessageResult;
 
 use bstr::{BStr, BString};
-#[cfg(not(unix))]
+#[cfg(all(not(unix), not(target_os = "wasi")))]
 use gix_error::ResultExt;
 
-#[cfg(not(unix))]
+#[cfg(unix)]
+use std::os::unix::ffi::{OsStrExt as _, OsStringExt as _};
+#[cfg(all(not(unix), target_os = "wasi"))]
+use std::os::wasi::ffi::{OsStrExt as _, OsStringExt as _};
+
+#[cfg(all(not(unix), not(target_os = "wasi")))]
 fn utf8_error() -> gix_error::Message {
     gix_error::validation("Could not convert to UTF8 or from UTF8 due to ill-formed input")
 }
@@ -49,24 +54,22 @@ pub fn try_into_bstr<'a>(path: impl Into<Cow<'a, Path>>) -> ExnMessageResult<Cow
     let path = path.into();
     let path_str = match path {
         Cow::Owned(path) => Cow::Owned({
-            #[cfg(unix)]
+            #[cfg(any(unix, target_os = "wasi"))]
             let p: BString = {
-                use std::os::unix::ffi::OsStringExt;
                 path.into_os_string().into_vec().into()
             };
-            #[cfg(not(unix))]
+            #[cfg(all(not(unix), not(target_os = "wasi")))]
             let p: BString = String::from_utf8(path.into_os_string().into_encoded_bytes())
                 .or_raise(utf8_error)?
                 .into();
             p
         }),
         Cow::Borrowed(path) => Cow::Borrowed({
-            #[cfg(unix)]
+            #[cfg(any(unix, target_os = "wasi"))]
             let p: &BStr = {
-                use std::os::unix::ffi::OsStrExt;
                 path.as_os_str().as_bytes().into()
             };
-            #[cfg(not(unix))]
+            #[cfg(all(not(unix), not(target_os = "wasi")))]
             let p: &BStr = std::str::from_utf8(path.as_os_str().as_encoded_bytes())
                 .or_raise(utf8_error)?
                 .as_bytes()
@@ -98,12 +101,11 @@ pub fn join_bstr_unix_pathsep<'a, 'b>(base: impl Into<Cow<'a, BStr>>, path: impl
 /// There are some potential Git versions and Windows installations which produce malformed UTF-16
 /// if certain emojis are in the path. It's as rare as it sounds, but possible.
 pub fn try_from_byte_slice(input: &[u8]) -> ExnMessageResult<&Path> {
-    #[cfg(unix)]
+    #[cfg(any(unix, target_os = "wasi"))]
     let p = {
-        use std::os::unix::ffi::OsStrExt;
         OsStr::from_bytes(input).as_ref()
     };
-    #[cfg(not(unix))]
+    #[cfg(all(not(unix), not(target_os = "wasi")))]
     let p = Path::new(std::str::from_utf8(input).or_raise(utf8_error)?);
     Ok(p)
 }
@@ -125,12 +127,11 @@ pub fn from_bstr<'a>(input: impl Into<Cow<'a, BStr>>) -> Cow<'a, Path> {
 /// Similar to [`try_from_bstr()`], but takes and produces owned data.
 pub fn try_from_bstring(input: impl Into<BString>) -> ExnMessageResult<PathBuf> {
     let input = input.into();
-    #[cfg(unix)]
+    #[cfg(any(unix, target_os = "wasi"))]
     let p = {
-        use std::os::unix::ffi::OsStringExt;
         std::ffi::OsString::from_vec(input.into()).into()
     };
-    #[cfg(not(unix))]
+    #[cfg(all(not(unix), not(target_os = "wasi")))]
     let p = PathBuf::from(String::from_utf8(input.into()).or_raise(utf8_error)?);
     Ok(p)
 }
