diff --git a/meson.build b/meson.build
index 90555e4..240143a 100644
--- a/meson.build
+++ b/meson.build
@@ -148,13 +148,18 @@ if host_machine.system() == 'windows'
     rc_data.set('API_VERSION_REVISION', dav1d_api_version_revision)
     rc_data.set('COPYRIGHT_YEARS', '2018-2026')
 else
+    if host_machine.system() == 'wasi'
+        thread_dependency = []
+        thread_compat_dep = []
+    else
     thread_dependency = dependency('threads')
     thread_compat_dep = []
+    endif
 
     rt_dependency = []
     if cc.has_function('clock_gettime', prefix : '#include <time.h>', args : test_args)
         have_clock_gettime = true
-    elif host_machine.system() not in ['darwin', 'ios', 'tvos']
+    elif host_machine.system() not in ['darwin', 'ios', 'tvos', 'wasi']
         rt_dependency = cc.find_library('rt', required: false)
         if not cc.has_function('clock_gettime', prefix : '#include <time.h>', args : test_args, dependencies : rt_dependency)
             error('clock_gettime not found')
@@ -219,7 +224,7 @@ if not cc.check_header('stdatomic.h')
     endif
 endif
 
-if host_machine.cpu_family().startswith('wasm')
+if host_machine.cpu_family().startswith('wasm') and host_machine.system() != 'wasi'
     # enable atomics + bulk-memory features
     stdatomic_dependencies += thread_dependency.partial_dependency(compile_args: true)
 endif
