diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6a663c..6c50271 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,11 +21,10 @@ option(CMARK_LIB_FUZZER "Build libFuzzer fuzzing harness" OFF)
 option(CMARK_FUZZ_QUADRATIC "Build quadratic fuzzing harness" OFF)
 
 if(CMARK_FUZZ_QUADRATIC)
-  set(FUZZER_FLAGS "-fsanitize=fuzzer-no-link,address -g")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FUZZER_FLAGS}")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${FUZZER_FLAGS}")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${FUZZER_FLAGS}")
-  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${FUZZER_FLAGS}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${FUZZ_LDFLAGS}")
+  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} $ENV{FUZZ_LDLFLAGS}")
 endif()
 
 add_subdirectory(src)
diff --git a/fuzz/CMakeLists.txt b/fuzz/CMakeLists.txt
index a9ed57a..3c5ca1c 100644
--- a/fuzz/CMakeLists.txt
+++ b/fuzz/CMakeLists.txt
@@ -9,8 +9,7 @@ macro(fuzzer name)
     add_executable(${name} ${name}.c)
     set_target_properties(${name}
           PROPERTIES
-          COMPILE_FLAGS "-fsanitize=fuzzer"
-          LINK_FLAGS "-fsanitize=fuzzer")
+	  LINK_FLAGS "$ENV{FUZZ_LDFLAGS}")
     if(CMARK_SHARED)
       target_link_libraries(${name} libcmark-gfm-extensions libcmark-gfm)
     elseif(CMARK_STATIC)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 84dd2a0..f49a868 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -221,10 +221,6 @@ endif()
 if(CMARK_LIB_FUZZER)
   set(FUZZ_HARNESS "cmark-fuzz")
   add_executable(${FUZZ_HARNESS} ../test/cmark-fuzz.c ${LIBRARY_SOURCES})
-  target_link_libraries(${FUZZ_HARNESS} "${CMAKE_LIB_FUZZER_PATH}")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize-coverage=trace-pc-guard")
-
-  # cmark is written in C but the libFuzzer runtime is written in C++ which
-  # needs to link against the C++ runtime. Explicitly link it into cmark-fuzz
-  set_target_properties(${FUZZ_HARNESS} PROPERTIES LINK_FLAGS "-lstdc++")
+  # target_link_libraries(${FUZZ_HARNESS} "${CMAKE_LIB_FUZZER_PATH}")
+  set_target_properties(${FUZZ_HARNESS} PROPERTIES LINK_FLAGS "$ENV{FUZZ_LDFLAGS}")
 endif()
