diff --git a/naga/fuzz/Cargo.toml b/naga/fuzz/Cargo.toml
index 0e63bc4a2..ecf5ae2b4 100644
--- a/naga/fuzz/Cargo.toml
+++ b/naga/fuzz/Cargo.toml
@@ -10,11 +10,11 @@ build = "build.rs"
 [package.metadata]
 cargo-fuzz = true
 
-[target.'cfg(not(any(target_arch = "wasm32", target_os = "ios")))'.dependencies]
+[dependencies]
 arbitrary = { workspace = true, features = ["derive"] }
 libfuzzer-sys.workspace = true
 
-[target.'cfg(not(any(target_arch = "wasm32", target_os = "ios", target_os = "visionos")))'.dependencies.naga]
+[dependencies.naga]
 workspace = true
 features = ["arbitrary", "spv-in", "wgsl-in", "glsl-in"]
 
diff --git a/naga/fuzz/build.rs b/naga/fuzz/build.rs
index 9ad00f53a..a69970d63 100644
--- a/naga/fuzz/build.rs
+++ b/naga/fuzz/build.rs
@@ -1,6 +1,6 @@
 fn main() {
     cfg_aliases::cfg_aliases! {
-        fuzzable_platform: { not(any(target_arch = "wasm32", target_os = "ios", all(windows, target_arch = "aarch64"))) },
+        fuzzable_platform: { not(false) },
     }
     // This cfg provided by cargo-fuzz
     println!("cargo::rustc-check-cfg=cfg(fuzzing)");
diff --git a/rust-toolchain.toml b/rust-toolchain.toml
deleted file mode 100644
index 04d37a72c..000000000
--- a/rust-toolchain.toml
+++ /dev/null
@@ -1,4 +0,0 @@
-[toolchain]
-channel = "1.92"
-components = ["rustfmt", "clippy"]
-targets = ["wasm32-unknown-unknown"]
