diff --git a/Cargo.toml b/Cargo.toml
index 32461ed..677d560 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,6 @@ members = [
     "lock-analyzer",
     "naga-cli",
     "naga",
-    "naga/fuzz",
     "naga/hlsl-snapshots",
     "player",
     "tests",
@@ -22,7 +21,7 @@ members = [
     "wgpu-types",
     "wgpu",
 ]
-exclude = []
+exclude = ["naga/fuzz"]
 default-members = [
     "benches",
     "examples/features",
@@ -30,7 +29,6 @@ default-members = [
     "lock-analyzer",
     "naga-cli",
     "naga",
-    "naga/fuzz",
     "naga/hlsl-snapshots",
     "player",
     "tests",
diff --git a/naga/Cargo.toml b/naga/Cargo.toml
index 4458405..0a28e01 100644
--- a/naga/Cargo.toml
+++ b/naga/Cargo.toml
@@ -74,7 +74,7 @@ hlsl-out-if-target-windows = []
 compact = []
 
 [dependencies]
-arbitrary = { version = "1.4", features = ["derive"], optional = true }
+arbitrary = { git = "https://github.com/rust-fuzz/arbitrary.git", rev = "ef80790c5bbcd24f342967e2388aa14f2c0d4a6b", features = ["derive"], optional = true }
 arrayvec.workspace = true
 bitflags.workspace = true
 bit-set.workspace = true
diff --git a/naga/fuzz/Cargo.toml b/naga/fuzz/Cargo.toml
index 5d8647f..e44f3e4 100644
--- a/naga/fuzz/Cargo.toml
+++ b/naga/fuzz/Cargo.toml
@@ -10,18 +10,18 @@ build = "build.rs"
 [package.metadata]
 cargo-fuzz = true
 
-[target.'cfg(not(any(target_arch = "wasm32", target_os = "ios")))'.dependencies]
+[dependencies]
 arbitrary = { version = "1.4.1", features = ["derive"] }
 # See https://github.com/rust-fuzz/libfuzzer/issues/126
 libfuzzer-sys = ">0.4.0,<=0.4.7"
 
-[target.'cfg(not(any(target_arch = "wasm32", target_os = "ios", target_os = "visionos")))'.dependencies.naga]
+[dependencies.naga]
 path = ".."
 version = "24.0.0"
 features = ["arbitrary", "spv-in", "wgsl-in", "glsl-in"]
 
 [build-dependencies]
-cfg_aliases.workspace = true
+cfg_aliases = "*"
 
 [[bin]]
 name = "spv_parser"
@@ -53,3 +53,7 @@ doc = false
 
 [lints.clippy]
 disallowed_types = "allow"
+
+# Prevent this from interfering with workspaces
+[workspace]
+members = ["."]
diff --git a/naga/fuzz/build.rs b/naga/fuzz/build.rs
index 9ad00f5..b7a9ae4 100644
--- a/naga/fuzz/build.rs
+++ b/naga/fuzz/build.rs
@@ -1,6 +1,6 @@
 fn main() {
     cfg_aliases::cfg_aliases! {
-        fuzzable_platform: { not(any(target_arch = "wasm32", target_os = "ios", all(windows, target_arch = "aarch64"))) },
+        fuzzable_platform: { not(false) },
     }
     // This cfg provided by cargo-fuzz
     println!("cargo::rustc-check-cfg=cfg(fuzzing)");
