diff --git a/CMakeLists.txt b/CMakeLists.txt
index aa73591..15c3862 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ set(CPACK_RESOURCE_FILE_README ${zlib_SOURCE_DIR}/README)
 # ============================================================================
 
 option(ZLIB_BUILD_TESTING "Enable Zlib Examples as tests" ON)
-option(ZLIB_BUILD_SHARED "Enable building zlib shared library" ON)
+option(ZLIB_BUILD_SHARED "Enable building zlib shared library" OFF)
 option(ZLIB_BUILD_STATIC "Enable building zlib static library" ON)
 option(ZLIB_BUILD_MINIZIP "Enable building libminizip contrib library" OFF)
 option(ZLIB_INSTALL "Enable installation of zlib" ON)
