diff --git a/jxl/src/frame/modular/transforms/squeeze.rs b/jxl/src/frame/modular/transforms/squeeze.rs
index 4a0e6ef..c839530 100644
--- a/jxl/src/frame/modular/transforms/squeeze.rs
+++ b/jxl/src/frame/modular/transforms/squeeze.rs
@@ -462,7 +462,9 @@ pub fn do_hsqueeze_step(
     buffers: &mut [&mut ModularChannel],
 ) {
     trace!("hsqueeze step in_avg: {in_avg:?} in_res: {in_res:?} in_next_avg: {in_next_avg:?}");
-    let out = buffers.first_mut().unwrap();
+    let Some(out) = buffers.first_mut() else {
+        return;
+    };
     // Shortcut: guarantees that row is at least 1px in the main loop
     if out.data.size().0 == 0 || out.data.size().1 == 0 {
         return;
@@ -661,7 +663,10 @@ pub fn do_vsqueeze_step(
     buffers: &mut [&mut ModularChannel],
 ) {
     trace!("vsqueeze step in_avg: {in_avg:?} in_res: {in_res:?} in_next_avg: {in_next_avg:?}");
-    let out = &mut buffers.first_mut().unwrap().data;
+    let Some(out_ch) = buffers.first_mut() else {
+        return;
+    };
+    let out = &mut out_ch.data;
     // Shortcut: guarantees that there at least 1 output row
     if out.size().1 == 0 || out.size().0 == 0 {
         return;
