Subject: [PATCH] bounds-check modular pipeline buffer indices
---
diff --git a/jxl/src/frame/modular/mod.rs b/jxl/src/frame/modular/mod.rs
index fec0399..6e4caea 100644
--- a/jxl/src/frame/modular/mod.rs
+++ b/jxl/src/frame/modular/mod.rs
@@ -433,7 +433,7 @@ impl FullModularImage {
             &section_buffer_indices,
             &mut buffer_info,
             modular_color_channels,
-        );
+        )?;
 
         #[cfg(feature = "tracing")]
         for (i, bi) in buffer_info.iter().enumerate() {
