Subject: [PATCH] return error on RCT grid kind mismatch
---
diff --git a/jxl/src/frame/modular/transforms/meta_apply.rs b/jxl/src/frame/modular/transforms/meta_apply.rs
index 2305805..1d9fb8c 100644
--- a/jxl/src/frame/modular/transforms/meta_apply.rs
+++ b/jxl/src/frame/modular/transforms/meta_apply.rs
@@ -304,7 +304,7 @@ pub fn make_grids(
     section_buffer_indices: &[Vec<usize>],
     buffer_info: &mut Vec<ModularBufferInfo>,
     color_channels: usize,
-) -> Vec<TransformStepChunk> {
+) -> Result<Vec<TransformStepChunk>> {
     // Initialize grid sizes, starting from coded channels.
     for i in section_buffer_indices[1].iter() {
         buffer_info[*i].grid_kind = ModularGridKind::Lf;
@@ -327,7 +327,9 @@ pub fn make_grids(
             } => {
                 let grid_in = buffer_info[buf_in[0]].grid_kind;
                 for i in 0..3 {
-                    assert_eq!(grid_in, buffer_info[buf_in[i]].grid_kind);
+                    if grid_in != buffer_info[buf_in[i]].grid_kind {
+                        return Err(Error::MixingDifferentChannels);
+                    }
                 }
                 for i in 0..3 {
                     buffer_info[buf_out[i]].grid_kind = grid_in;
@@ -712,5 +714,5 @@ pub fn make_grids(
 
     trace!(?grid_transform_steps, ?buffer_info);
 
-    grid_transform_steps
+    Ok(grid_transform_steps)
 }
