From a17306eaac72a6db11fb902430e9f3709bb2c986 Mon Sep 17 00:00:00 2001
Date: Wed, 3 Jan 2024 17:55:45 +0100
Subject: [PATCH] HACK: remove patch.crates-io

---
 fuzz/Cargo.toml | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/fuzz/Cargo.toml b/fuzz/Cargo.toml
index 69221ca..df59b2e 100644
--- a/fuzz/Cargo.toml
+++ b/fuzz/Cargo.toml
@@ -33,10 +33,6 @@ path = "fuzz_targets/x509_with_mutator.rs"
 test = false
 doc = false
 
-[patch.crates-io]
-# der-parser = { path="../../der-parser" }
-oid-registry = { git="https://github.com/rusticata/oid-registry" }
-
 [[bin]]
 name = "certreq"
 path = "fuzz_targets/certreq.rs"
-- 
2.46.0

