diff --git a/fuzz/Cargo.toml b/fuzz/Cargo.toml
index 2076fc8..f403f08 100644
--- a/fuzz/Cargo.toml
+++ b/fuzz/Cargo.toml
@@ -36,7 +36,7 @@ doc = false
 [patch.crates-io]
 # der-parser = { path="../../der-parser" }
 asn1-rs = { git="https://github.com/rusticata/asn1-rs", branch = "master" }
-oid-registry = { git="https://github.com/rusticata/oid-registry", branch="oid-registry-0.9" }
+# oid-registry = { git="https://github.com/rusticata/oid-registry", branch="oid-registry-0.9" }
 
 [[bin]]
 name = "certreq"
