diff --git a/crates/ty/Cargo.toml b/crates/ty/Cargo.toml
index e9bc06c..4a66acf 100644
--- a/crates/ty/Cargo.toml
+++ b/crates/ty/Cargo.toml
@@ -22,7 +22,7 @@ ruff_db = { workspace = true, features = ["os", "cache", "junit"] }
 ruff_diagnostics = { workspace = true }
 ruff_ranged_value = { workspace = true }
 ty_combine = { workspace = true }
-ty_project = { workspace = true, features = ["zstd", "junit"] }
+ty_project = { workspace = true, features = ["junit"] }
 ty_python_semantic = { workspace = true, features = ["serde"] }
 ty_server = { workspace = true }
 ty_static = { workspace = true }
diff --git a/crates/ty_project/Cargo.toml b/crates/ty_project/Cargo.toml
index de0e541..b9fdcf6 100644
--- a/crates/ty_project/Cargo.toml
+++ b/crates/ty_project/Cargo.toml
@@ -65,7 +65,7 @@ ruff_db = { workspace = true, features = ["os", "testing"] }
 insta = { workspace = true, features = ["redactions", "ron"] }
 
 [features]
-default = ["zstd"]
+default = []
 deflate = ["ty_vendored/deflate"]
 schemars = [
     "dep:schemars",
