diff --git a/crates/ty/Cargo.toml b/crates/ty/Cargo.toml
index 12bbce58c6..49093a77b4 100644
--- a/crates/ty/Cargo.toml
+++ b/crates/ty/Cargo.toml
@@ -20,7 +20,7 @@ doctest = false
 ruff_db = { workspace = true, features = ["os", "cache", "junit"] }
 ruff_diagnostics = { workspace = true }
 ty_combine = { workspace = true }
-ty_project = { workspace = true, features = ["zstd", "junit"] }
+ty_project = { workspace = true, features = ["junit"] }
 ty_python_semantic = { workspace = true, features = ["serde"] }
 ty_server = { workspace = true }
 ty_static = { workspace = true }
diff --git a/crates/ty_project/Cargo.toml b/crates/ty_project/Cargo.toml
index 8a3b32aea9..6136c11e8d 100644
--- a/crates/ty_project/Cargo.toml
+++ b/crates/ty_project/Cargo.toml
@@ -60,7 +60,7 @@ ruff_db = { workspace = true, features = ["testing"] }
 insta = { workspace = true, features = ["redactions", "ron"] }
 
 [features]
-default = ["zstd"]
+default = []
 deflate = ["ty_vendored/deflate"]
 schemars = [
     "dep:schemars",
diff --git a/crates/ty_vendored/Cargo.toml b/crates/ty_vendored/Cargo.toml
index a9b70a07ec..4509fd358b 100644
--- a/crates/ty_vendored/Cargo.toml
+++ b/crates/ty_vendored/Cargo.toml
@@ -18,7 +18,7 @@ zip = { workspace = true }
 [build-dependencies]
 path-slash = { workspace = true }
 walkdir = { workspace = true }
-zip = { workspace = true, features = ["zstd", "deflate"] }
+zip = { workspace = true, features = ["deflate"] }
 
 [dev-dependencies]
 walkdir = { workspace = true }
diff --git a/crates/ty_vendored/build.rs b/crates/ty_vendored/build.rs
index 5c65c09848..81acaf453a 100644
--- a/crates/ty_vendored/build.rs
+++ b/crates/ty_vendored/build.rs
@@ -34,7 +34,7 @@ fn write_zipped_typeshed_to(writer: File) -> ZipResult<File> {
     // architecture of the system running the build script and not the architecture of the build-target.
     // That's why we use the `TARGET` environment variable here.
     let method = if cfg!(feature = "zstd") {
-        CompressionMethod::Zstd
+        CompressionMethod::Stored
     } else if cfg!(feature = "deflate") {
         CompressionMethod::Deflated
     } else {
